mentat/code_context.py

```diff
@@ -271,37 +273,133 @@ def _get_all_features(
+        """Remove code from the context
+
+        # NOTE: should this be the intended behavior?
+        Code Features are removed from the context and are excluded from any future `include` calls (unless explicity specified).
+        For paths that are files or file intervals, if they don't already exist in the Code Context an Exception is thrown.
+        For file intervals, if the interval doesn't *exactly match* any intervals in the code context, nothing is removed.
+
+        Args:
+            `path`: can be a relative or absolute file path, file interval path, directory, or glob pattern.
+
+        Return:
+            A set of paths that have been successfully removed from the context
+        """
+        session_context = SESSION_CONTEXT.get()
+
+        path = Path(path)
+        validated_path = validate_and_format_path(
+            path, session_context.cwd, check_for_text=False
+        )
+
+        excluded_paths: Set[Path] = set()
+        try:
+            # file
+            if validated_path.is_file():
+                if validated_path not in self.include_files:
+                    session_context.stream.send(
+                        f"Path {validated_path} not in context", color="light_red"
+                    )
+                    return excluded_paths
+                excluded_paths.add(validated_path)
+                del self.include_files[validated_path]
+            # file interval
+            elif len(str(validated_path).rsplit(":", 1)) > 1:
+                interval_path, interval_str = str(validated_path).rsplit(":", 1)
+                interval_path = Path(interval_path)
+                if interval_path not in self.include_files:
+                    session_context.stream.send(
+                        f"Path {interval_path} not in context", color="light_red"
+                    )
+                    return excluded_paths
+                intervals = parse_intervals(interval_str)
+                included_code_features: List[CodeFeature] = []
+                for code_feature in self.include_files[interval_path]:
+                    if code_feature.interval in intervals:
+                        excluded_path = f"{interval_path}:{code_feature.interval.start}-{code_feature.interval.end}"
+                        excluded_paths.add(Path(excluded_path))
+                    else:
+                        included_code_features.append(code_feature)
+                self.include_files[interval_path] = included_code_features
+            # directory
+            elif validated_path.is_dir():
+                if validated_path not in self.include_files:
+                    session_context.stream.send(
+                        f"Directory path {validated_path} not in context",
+                        color="light_red",
+                    )
+                    return excluded_paths
+                excluded_paths.add(validated_path)
+                del self.include_files[validated_path]
```

Comment:
Need to fix this. All paths inside of `validated_path` (a directory) should be removed *instead* of checking if `validated_path` is in `self._include_files` (which is impossible)

Ty for @PCSwingle for finding this earlier in https://github.com/AbanteAI/mentat/pull/246#discussion_r1385344004